### PR TITLE
Feature/helper code

### PR DESCRIPTION
**Pull request by ObiWahn**

### CODE REVIEW

File: lib/Basics/Exceptions.h [hunk 1]
```diff
@@ -58,6 +58,25 @@
 #define THROW_ARANGO_EXCEPTION_MESSAGE(code, message) \
   throw arangodb::basics::Exception(code, message, __FILE__, __LINE__)
 
+/// @brief throws an arango result if the result fails
+#define THROW_ARANGO_EXCEPTION_IF_FAIL(reSUlt) \
```
Comment (jsteemann) [inline lib/Basics/Exceptions.h]: why `reSUlt` and not `result`?

File: lib/Basics/Exceptions.h [hunk 2]
```diff
@@ -58,6 +58,25 @@
...
+#define CATCH_TO_RESULT_WITH_CODE(result, errorCode) \
+  catch (arangodb::basics::Exception const& e) { \
+    (result).reset(e.code(),e.message()); \
+  } catch (std::exception const& e) { \
+    (result).reset((errorCode),e.what()); \
+  } catch (...) { \
+    (result).reset((errorCode)); \
+  }
+
+#define CATCH_TO_RESULT(result) CATCH_TO_RESULT_WITH_CODE((result),TRI_ERROR_INTERNAL)
```
Comment (jsteemann) [inline lib/Basics/Exceptions.h]: Can we rename macro `CATCH_TO_RESULT_WITH_CODE` to just `CATCH_RESULT`, and remove macro `CATCH_TO_RESULT` altogether? I don't think we need two macros.

File: lib/Basics/ScopedTimer.h [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Jan Christoph Uhde
+////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
```
Comment (jsteemann) [inline lib/Basics/ScopedTimer.h]: `#pragma once` is disliked by @fceller. Please use just the regular include guards.

File: lib/Basics/ScopedTimer.h [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+public:  // defines
+    using hclock    = std::chrono::high_resolution_clock;
+    using hrcStrVec = std::vector<std::pair<hclock::time_point,std::string>>;
+    using intStrVec = std::vector<std::pair<int,std::string>>;
+
+private:  // variables
+    hrcStrVec _timePoints;
+    std::function<void(intStrVec)> _callback;
+    bool _enabled;
+    bool _add_dtor_entry;
```
Comment (jsteemann) [inline lib/Basics/ScopedTimer.h]: note that we use camelCase and no snake_case as part of our coding conventions.

File: lib/Logger/LogMacros.h [hunk 5]
```diff
@@ -106,6 +106,13 @@
                                   << arangodb::Logger::FILE(__FILE__)    \
                                   << arangodb::Logger::FUNCTION(__FUNCTION__))
 
+////////////////////////////////////////////////////////////////////////////////
+/// @brief logs a message for debugging during development
+////////////////////////////////////////////////////////////////////////////////
+
+#define LOG_DEVEL LOG_TOPIC(ERR,Logger::FIXME) << "###### "
```
Comment (jsteemann) [inline lib/Logger/LogMacros.h]: Why not use `Logger::DEVEL` as the topic here?
I am a bit worried that these macros will make us add even more log garbage than we already have. How can we prevent that from happening?

Comment (ObiWahn) [reply lib/Logger/LogMacros.h]: I had devel first as log topic but it is unfortunately not visible in the default configuration.

File: lib/Basics/Exceptions.h (hunk 1 above)
Comment (ObiWahn) [reply lib/Basics/Exceptions.h]: There was a problem with the macro expansion that could be fixed by slightly different spelling of the parameter.

File: lib/Basics/ScopedTimer.h (hunk 3 above)
Comment (ObiWahn) [reply lib/Basics/ScopedTimer.h]: It has both the pragma and regular include guards. If the prama works it is faster in skipping files and if the compiler does not support once or a file has been copied to another location the include guards will enforce the correct behavior.

File: lib/Basics/ScopedTimer.h (hunk 4 above)
Comment (ObiWahn) [reply lib/Basics/ScopedTimer.h]: I forgot to change every variable when i imported the code form my repo. I'll fix the internal stuff as well.

File: lib/Basics/Exceptions.h (hunk 2 above)
Comment (ObiWahn) [reply lib/Basics/Exceptions.h]: Sure that could be done - It is probably better to force people to think about the correct default error code.

File: lib/Logger/LogMacros.h (hunk 5 above)
Comment (ObiWahn) [reply lib/Logger/LogMacros.h]: I have added this marco in the hope that I will be able to remove all my logging messages. Without forgetting any of them. The Logger::FIXME is all over the place and it is more writing than LOG_DEVEL. When a feature is merged back into devel the definition should be the only remaining LOG_DEVEL in the code base.

File: lib/Basics/ScopedTimer.h (hunk 3 above)
Comment (jsteemann) [reply lib/Basics/ScopedTimer.h]: @fceller will not merge it with `#pragma once`.

**Pull request merged**